frontend/src/components/Login.js

```diff
@@ -0,0 +1,179 @@
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2)
+  }
+}));
+
+const Login = () => {
+  const classes = useStyles();
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [mode, setMode] = useState(null);
+
+  const accessToken = useSelector((store) => store.user.accessToken);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  useEffect(() => {
+    // redirect user to '/' path
+    // everytime access token is changed, it will be pushed '/'
+    console.log("Checking access token", accessToken);
+    if (accessToken) {
+      history.push("/questions");
+    }
+  }, [accessToken, history]);
+
+  const onFormSubmit = (e) => {
+    e.preventDefault();
+
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify({ username, password })
+    };
+    fetch(API_URL(mode), options)
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        if (data.success) {
+          // updating dispatch at the same time
+          batch(() => {
+            dispatch(user.actions.setUsername(data.username));
+            dispatch(user.actions.setAccessToken(data.accessToken));
+            dispatch(user.actions.setErrors(null));
+          });
+        } else {
+          dispatch(user.actions.setErrors(data));
+        }
+      })
+      .catch();
+  };
+
+  return (
+    <div className={classes.background}>
+      <div className={classes.wrapper}>
+        <Typography className={classes.header}>Are you feeling blue..? </Typography>
+        <Typography className={classes.paragraph}>
+          Sign up or use existing account to know the secret how you can feel better{" "}
+        </Typography>
+      </div>
+      <Container component="main" maxWidth="xs">
+        <div className={classes.paper}>
+          <Avatar className={classes.avatar}>{/* <LockOutlinedIcon /> */}</Avatar>
+          <Typography component="h1" variant="h5">
+            Sign in
+          </Typography>
+          <form onSubmit={onFormSubmit}>
+            <TextField
+              variant="outlined"
+              margin="normal"
+              required
+              fullWidth
+              id="username"
+              label="Username"
+              name="username"
+              autoComplete="email"
+              autoFocus
+              type="text"
+              value={username}
+              onChange={(e) => setUsername(e.target.value)} />
+            <TextField
+              variant="outlined"
+              margin="normal"
+              required
+              fullWidth
+              name="password"
+              label="Password"
+              type="password"
+              id="password"
+              autoComplete="current-password"
+              type="password"
+              value={password}
+              onChange={(e) => setPassword(e.target.value)} />
```

Comment:
Maybe it would be better to create the external component with configurations of your components? it would make your code easier to read.

Reply:
Ah yes, this could be nice. We are trying to accomplish this for the future projects (sending in values as props)  Can we keep it like this for now?